force-app/main/default/classes/SOQL.cls

```diff
@@ -85,9 +100,17 @@ public with sharing class SOQL {
      * @return A map of Task Ids to Task records.
      * Update the corresponding test class and testGetTasksDueNextWeek method to cover this method
      */
-    //[public | private ] [static] data_type method_name (input parameters) {
-    //  The body of the method
-    //}
+    public static Map<Id, Task> getTasksDueNextWeek() {
+        Date nextWeek = Date.today().addDays(7);
+        List<Task> tasks = [SELECT Id, Subject, ActivityDate FROM Task WHERE ActivityDate <= :nextWeek];
+        // Create map to store the results
+        Map<Id, Task> taskMap = new Map<Id, Task>();
+        for (Task task : tasks) {
+            taskMap.put(task.Id, task);
+        }
+        
+        return taskMap;
```

Comment:
This is the ideal sue case of the special Map<Id, SObject> constructor or the SOQL for loop.